File: 2211-KRadiusSubarrayAverages/2211-KRadiusSubarrayAverages.cs
// Last updated: 10/8/2025, 9:48:09 PM
public class Solution {
public int[] GetAverages(int[] nums, int k)
{
    if (k == 0)
    {
        return nums;
    }
    int windowSize = 2 * k + 1;
    int n = nums.Length;
    int[] averages = Enumerable.Repeat(-1, n).ToArray();


    if (windowSize > n)
    {
        return averages;
    }
    long[] prefix = new long[n + 1];
    for (int i = 0; i < n; i++)
    {
        prefix[i + 1] = prefix[i] + nums[i];
    }

    for (int i = k; i < (n - k); ++i)
    {
        int leftBound = i - k, rightBound = i + k;
        long subArraySum = prefix[rightBound + 1] - prefix[leftBound];
        int average = Convert.ToInt32(subArraySum / windowSize);
        averages[i] = average;
    }

        return averages;


    }
}

File: 2211-KRadiusSubarrayAverages/2211-KRadiusSubarrayAverages.py
# Last updated: 10/8/2025, 9:48:04 PM
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        
        n=len(nums)
        res=[-1]*n
        if 2*k+1>n:
            return res
        window_sum=sum(nums[:2*k+1])
        res[k]=window_sum//(2*k+1)
        for i in range(k+1,n-k):
            window_sum=window_sum-nums[i-k-1]+nums[i+k]
            res[i]=window_sum//(2*k+1)
        return res